miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/context/PrometheusCollectionFilter.java

```diff
@@ -0,0 +1,246 @@
+package uk.ac.bbsrc.tgac.miso.webapp.context;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Locale;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.web.filter.GenericFilterBean;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Histogram;
+import uk.ac.bbsrc.tgac.miso.core.util.CoverageIgnore;
+
+/**
+ * Collect per-URL performance statistics and export them to Prometheus
+ */
+@SuppressWarnings("deprecation")
```

Comment:
I see 3 deprecated methods used - encodeUrl(String), encodeRedirectUrl(String), and setStatus(int, String).

Looking at the HttpServletResponse javadoc: https://javaee-spec.java.net/nonav/javadocs/javax/servlet/http/HttpServletResponse.html

encodeUrl and encodeRedirectUrl can be changed to encodeURL and encodeRedirectURL, which do the same thing (deprecated to change method names...)

setStatus:

> [deprecated] due to ambiguous meaning of the message parameter. To set a status code use setStatus(int), to send an error with a description use sendError(int, String). Sets the status code and message for this response.

Since you're just forwarding the call to the wrapped response object, changing this may not be reasonable, but the SuppressWarnings annotation could be moved to this method alone instead of the entire class